### ISSUE TITLE
Error in processPopulate if different components in a dynamic zone share the same attribute name

### ISSUE THREAD

**Issue by tilman** [strapi/strapi#13220]
### Required System information

- Node.js version: v14.18.2
- NPM version: yarn 1.22.17
- Strapi version: 4.0.5
- Database: psql
- Operating system: mac

### Describe the bug

If a dynamic zone includes two different components and both of the components have a field with the same name but one is from a primitive type (string, boolean etc.) and the other from a relation type (relation or media) then strapi server throws the following error as soon as both components are added in the dynamic zone:
```
[2022-05-02 15:54:00.041] error: Invalid populate field. Expected a relation, got string
Error: Invalid populate field. Expected a relation, got string
    at Object.processPopulate (.../node_modules/@strapi/database/lib/query/helpers/populate.js:83:13)
```
### Steps to reproduce the behavior

Create the following content type:
![image](https://user-images.githubusercontent.com/5111431/166246502-ae491097-8782-4189-9c77-92b9cc1f7258.png)
![image](https://user-images.githubusercontent.com/5111431/166246537-f2de7454-b56a-4ab0-a45a-020dd4ea1f42.png)

Create an instance of it and click save:
![image](https://user-images.githubusercontent.com/5111431/166246650-e8472c16-26bd-4dcc-9bb5-9bed4d9a2e1b.png)

### Expected behavior
The check should not throw an error if different components in a dynamic zone share the same attribute name

**Comment by derrickmehaffy** [comment]
> Strapi version: 4.0.5

Please test on the latest version (v4.1.9 as of writing) as you are quite a few versions behind. If the issue still occurs on the latest version, rely back and we will reopen.

**Comment by yuriyglushakov** [comment]
Cought same issue on Strapi v. 4.3.4

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report, 

- questions should be directed to [our forum](https://forum.strapi.io) or our [Discord](https://discord.strapi.io). 
- feature requests should be directed to our [feedback and feature request database](https://feedback.strapi.io)

Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Comment by tilman** [comment]
> Hello,
> 
> I see you are wanting to ask a question that is not really a bug report,
> 
> * questions should be directed to [our forum](https://forum.strapi.io) or our [Discord](https://discord.strapi.io).
> * feature requests should be directed to our [feedback and feature request database](https://feedback.strapi.io)
> 
> Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

For me it doesn‘t look like a question. He just tested it on the latest version as you have asked. So I think we can reopen the issue?

**Comment by derrickmehaffy** [comment]
> > Hello,
> > I see you are wanting to ask a question that is not really a bug report,
> > 
> > * questions should be directed to [our forum](https://forum.strapi.io) or our [Discord](https://discord.strapi.io).
> > * feature requests should be directed to our [feedback and feature request database](https://feedback.strapi.io)
> > 
> > Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
> 
> For me it doesn‘t look like a question. He just tested it on the latest version as you have asked. So I think we can reopen the issue?

apologizes you are right, I missed the context at the top.

**Comment by derrickmehaffy** [comment]
Raising this to high as it directly impacts migration per a user on Discord:

![image](https://user-images.githubusercontent.com/8593673/185683759-7cd2d15c-f1f0-4c6e-b307-cc94210fe727.png)

**Comment by Convly** [comment]
Re-opening this one as even if the feature allowing to fix this issue has been released, the admin does not reflect the changes yet.

**Comment by Marc-Roig** [comment]
It seems this bug is fixed, I can not reproduce it (using 4.5.6).
Seems we are populating correctly each component rn.

I am closing the issue and we can open it again if something related happens again.
